eclipse-scout-core/test/form/fields/smartfield/ProposalFieldSpec.js

```diff
@@ -157,4 +157,139 @@ describe('ProposalField', () => {
     field._acceptInput(false, 'Bar', false, true, null);
     expect(field.getErrorStatus).toBeUndefined();
   });
+
+  describe('acceptInput-event contains correct data', () => {
+    let modelAdapter, spy;
+
+    beforeEach(() => {
+      jasmine.clock().uninstall();
+
+      linkWidgetAndAdapter(field, 'ProposalFieldAdapter');
+      modelAdapter = field.modelAdapter;
+      spy = spyOn(modelAdapter, '_send');
+
+      field.setLookupCall(scout.create('StaticLookupCall', {
+        session: field.session,
+        data: [
+          ['foo', 'foo'],
+          ['bar', 'bar']
+        ]
+      }));
+    });
+
+    // if the value is changed it needs to be sent to the server
+    // test this for various inputs (custom text and lookupRow selection) in 'normal'/touch mode and with/without an error status present
+    testCases().forEach(testCase => {
```

Comment:
It is interesting that it is possible to create custom test cases, but it looks quite complicated and is probably not easy to debug if one case fails... Why don't you just create a real test method for each case? I mean, the test methods would only contain a call to testProposalFieldInputs, wouldn't they?